lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -741,19 +738,16 @@ mod tests {
 
         let top = 5;
         let query = random_vector(&mut rng, dim);
-        let processed_query = <M as Metric<VectorElementType>>::preprocess(query.clone());
+        let scorer = vector_holder.scorer(query.clone());
         let mut reference_top = FixedLengthPriorityQueue::new(top);
-        for idx in 0..vector_holder.vectors.len() as PointOffsetType {
-            let vec = &vector_holder.vectors.get(idx as VectorOffsetType);
-            reference_top.push(ScoredPointOffset {
-                idx,
-                score: M::similarity(vec, &processed_query),
-            });
+        for idx in 0..vector_holder.storage().total_vector_count() as PointOffsetType {
+            let score = scorer.score_point(idx);
+            reference_top.push(ScoredPointOffset { idx, score });
```

Comment:
Didn't we lose preprocessing here (and in other places)?

Reply:
`vector_holder.scorer` (aka `TestRawScorerProducer::scorer`) goes into `FilteredScorerImpl` then into `new_raw_scorer` then all the way into `MetricQueryScorer::new()` which preprocesses the query.

So, no.